node/api/wallet.go

```diff
@@ -521,10 +548,28 @@ func (api *API) walletTransactionsHandler(w http.ResponseWriter, req *http.Reque
 	}
 	unconfirmedTxns := api.wallet.UnconfirmedTransactions()
 
-	WriteJSON(w, WalletTransactionsGET{
-		ConfirmedTransactions:   confirmedTxns,
-		UnconfirmedTransactions: unconfirmedTxns,
-	})
+	confirmedPaginationRequest := GetPaginationRequest(req, "confirmedTransactionsStart", "confirmedTransactionsLimit")
+	unconfirmedPaginationRequest := GetPaginationRequest(req, "unconfirmedTransactionsStart", "unconfirmedTransactionsLimit")
+
+	if !confirmedPaginationRequest.HasPaginationQuery || !unconfirmedPaginationRequest.HasPaginationQuery || confirmedTxns == nil || unconfirmedTxns == nil {
+		WriteJSON(w, WalletTransactionsGET{
+			ConfirmedTransactions:   confirmedTxns,
+			UnconfirmedTransactions: unconfirmedTxns,
+		})
+	} else {
```

Comment:
We try to avoid nested `if` statements. You can get rid of the `else` by returning after the call to `WriteJSON`. In the API we usually don't do anything after calling `WriteJSON` or `WriteError` which means you can usually simplify your code by returning right afterwards.